### PR TITLE
network: can: add support for listen-only mode

### LINKED ISSUE #15129: Feature Request: Allow "listen-only" mode for socket can interfaces
The support for CAN (Controller Area Network) devices was added to systemd in 2018 (as per #9185). Currently, there are several supported options like CAN.BitRate, CAN.SamplePoint, CAN.RestartSec, however the "listen-only" mode is not yet implemented. 

Listen-only is important to sniff on CAN networks without letting the other bus members know that a third party device is even attached, as it would be with a simple ACK message sent from that device.

Unfortunately, I'm not deep enough in that code for that I could do the required code changes on my own.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #15129.

### CODE REVIEW

**Comment by yuwata** [comment]
> May be add "Listen-only is important to sniff on CAN networks without letting the other bus members know that a third party device is even attached, as it would be with a simple ACK message sent from that device."

@ssahani Thank you for the review. I've added more sentences based on yours. PTAL.

**Comment by ssahani** [comment]
LGTM

File: man/systemd.network.xml [hunk 1]
```diff
@@ -2317,6 +2317,16 @@
             the value of a received bit by majority rule. When unset, the kernel's default will be used.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>ListenOnly=</varname></term>
+          <listitem>
+            <para>Takes a boolean. When <literal>yes</literal>, listen-only mode is enabled. When the
+            interface is in listen-only mode, the interface does not transmit CAN frames or not send
```
Comment (boucman) [inline man/systemd.network.xml]: " or not send" needs rewording

Comment (yuwata) [reply man/systemd.network.xml]: How about "the interface neither transmit CAN frames or not send ACK bit."?

Comment (boucman) [reply man/systemd.network.xml]: "nor send ACK bit" sounds better to me...

Comment (yuwata) [reply man/systemd.network.xml]: OK. Will update. Thanks.

**Comment by trabant-rgb** [comment]
Wow... that was incredibly fast... works well for me. Thanks for taking care!

**Pull request merged**